# Projekt IAU  
## 1. časť - Prieskumná analýza 
### Vypracovali: Martin Kukučka, Martin Jankuliak

In [39]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
import scipy.stats as stats
import glob

import datetime
from datetime import datetime,date
from datetime import timedelta
from dateutil.relativedelta import relativedelta

import json
import re
import dateutil.parser as parser
import statsmodels.api as sm

import scipy as sc
from pandas import read_csv


### Načítanie dát

In [41]:
df_other_train = read_csv("data/other_train.csv")
df_personal_train = read_csv("data/personal_train.csv")

## Základný opis dát spolu s ich charakteristikami 

### Dataset other_train.csv

Pre predstavu tu je ukážka datasetu

In [53]:
df_other_train.head()

Unnamed: 0             name  \
0           0    Robert Plante   
1           1     Richard Huey   
2           2     Joseph Sable   
3           3  Gregory Preston   
4           4       Mike Riley   

                                             address                 race  \
0  3594 Johnson Green Suite 277\r\nJohnsonport, V...                White   
1             350 Lewis Haven\r\nAngelaton, IN 39486   Asian-Pac-Islander   
2  22509 Adrian Canyon Suite 799\r\nEast Timothyt...                White   
3           2674 Daryl Island\r\nLake Sean, TN 89363                White   
4  023 Joseph Estate Suite 799\r\nLake Andrewvill...                White   

        marital-status          occupation pregnant  education-num  \
0              Widowed       Other-service        f            5.0   
1        Never-married   Handlers-cleaners        f           13.0   
2        Never-married     Exec-managerial        f            9.0   
3   Married-civ-spouse     Exec-managerial        f           13.0   
4   Married-civ-spouse   Handlers-cleaners        f            9.0   

     relationship  capital-gain   education    fnlwgt  class  income  \
0       Unmarried           0.0         9th  220019.0    1.0   <=50K   
1   Not-in-family           0.0   Bachelors  196899.0    0.0   <=50K   
2       Unmarried           0.0     HS-grad  301591.0    0.0   <=50K   
3         Husband           0.0   Bachelors  310320.0    1.0    >50K   
4         Husband           0.0     HS-grad  237943.0    0.0   <=50K   

                                        medical_info  native-country  \
0  {'mean_glucose':'106.71875','std_glucose':'48....   United-States   
1  {'mean_glucose':'118.609375','std_glucose':'na...           Haiti   
2  {'mean_glucose':'129.4609375','std_glucose':'5...   United-States   
3  {'mean_glucose':'30.1015625','std_glucose':'37...   United-States   
4  {'mean_glucose':'124.0390625','std_glucose':'4...   United-States   

   hours-per-week  capital-loss workclass  
0             9.0           0.0   Private  
1            50.0           0.0   Private  
2            35.0           0.0   Private  
3            40.0           0.0   Private  
4            60.0           0.0   private

Pomocou tohto príkazu zistíme počet riadkov

In [54]:
df_other_train.shape[0]

3983

A teraz počet atribútov pre tento dataset

In [55]:
df_other_train.shape[1]

19

Typy atribútov zistíme týmto príkazom

In [56]:
df_other_train.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3983 entries, 0 to 3982
Data columns (total 19 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   Unnamed: 0      3983 non-null   int64  
 1   name            3983 non-null   object 
 2   address         3983 non-null   object 
 3   race            3691 non-null   object 
 4   marital-status  3970 non-null   object 
 5   occupation      3961 non-null   object 
 6   pregnant        3969 non-null   object 
 7   education-num   3961 non-null   float64
 8   relationship    3954 non-null   object 
 9   capital-gain    3969 non-null   float64
 10  education       3972 non-null   object 
 11  fnlwgt          3968 non-null   float64
 12  class           3960 non-null   float64
 13  income          3972 non-null   object 
 14  medical_info    3966 non-null   object 
 15  native-country  3966 non-null   object 
 16  hours-per-week  3966 non-null   float64
 17  capital-loss    3973 non-null   f

Zistile sme, že tento dataset je zložený z atribútov:
- 6x float
- 1x int
- 12x object

Ďalej tu vidíme aj počet riadkov: 3983 ("RangeIndex: 3983 entries"),
a počet stĺpcov: 19 ("Data columns (total 19 columns)")

Nakoniec sa pozrieme na desktiptívne štatistiky.

In [58]:
df_other_train.describe()

Unnamed: 0  education-num  capital-gain         fnlwgt        class  \
count  3983.00000    3961.000000   3969.000000    3968.000000  3960.000000   
mean   1991.00000      10.052764   1134.530108  189963.996976     0.256313   
std    1149.93739       2.520019   8067.028170  102327.438172     0.436652   
min       0.00000       1.000000      0.000000   20308.000000     0.000000   
25%     995.50000       9.000000      0.000000  119475.250000     0.000000   
50%    1991.00000      10.000000      0.000000  180585.500000     0.000000   
75%    2986.50000      12.000000      0.000000  236173.000000     1.000000   
max    3982.00000      16.000000  99999.000000  795830.000000     1.000000   

       hours-per-week  capital-loss  
count     3966.000000   3973.000000  
mean        40.203984     72.926252  
std         12.317838    360.910859  
min          1.000000      0.000000  
25%         40.000000      0.000000  
50%         40.000000      0.000000  
75%         45.000000      0.000000  
max         99.000000   2824.000000

V tejto tabuľke vidíme ako sú: počet záznamov, priemer, štandardnú odchýlku, minimum, prvý kvartil, druhý kvartil (medián), tretí kvartil, maximum, pre všetky atribúty, ktoré obsahujú numerické hodnoty.

### Dataset personal_train.csv

Rovnako vykonáme opis dát aj pre druhý dataset

Počet riadkov

In [46]:
df_personal_train.shape[0]

3933

Počet atribútov

In [47]:
df_personal_train.shape[1]

6

Typy atribútov

In [62]:
df_personal_train.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3933 entries, 0 to 3932
Data columns (total 6 columns):
 #   Column         Non-Null Count  Dtype 
---  ------         --------------  ----- 
 0   Unnamed: 0     3933 non-null   int64 
 1   name           3933 non-null   object
 2   address        3933 non-null   object
 3   age            3933 non-null   int64 
 4   sex            3933 non-null   object
 5   date_of_birth  3933 non-null   object
dtypes: int64(2), object(4)
memory usage: 184.5+ KB


Zistile sme, že tento dataset je zložený z atribútov:
- 2x int
- 4x object

Ďalej tu vidíme aj počet riadkov: 3933 ("RangeIndex: 3933 entries"),
a počet stĺpcov: 6 ("Data columns (total 6 columns)")

In [ ]:
A Nakoniec desktiptívne štatistiky.

In [63]:
df_personal_train.describe()

Unnamed: 0          age
count  3933.000000  3933.000000
mean   1966.000000    51.816934
std    1135.503633    11.814803
min       0.000000    -1.000000
25%     983.000000    44.000000
50%    1966.000000    52.000000
75%    2949.000000    60.000000
max    3932.000000   113.000000